src/terminal/parser/OutputStateMachineEngine.cpp

```diff
@@ -1624,7 +1619,7 @@ bool OutputStateMachineEngine::_GetOscSetColorTable(const std::wstring_view stri
 
     // First try to get the table index, a number between [0,256]
     size_t current = 0;
-    for (size_t i = 0; i < 4; i++)
+    for (size_t i = 0; i < string.size(); i++)
     {
         const wchar_t wch = string.at(current);

```

Comment:
Not a big deal, but I think you could probably make this `while (current < string.size())` and get rid of that `i` variable - they're effectively the same value. It just seems a bit misleading checking that `i` is less than the string size, but then using `current` to access the string.